src/common/extensionMessaging.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import * as net from "net";
+
+/**
+ * Pipe path used for communicating with the server.
+ */
+let WIN_ServerPipePath = "\\\\?\\pipe\\vscodereactnative";
+let UNIX_ServerPipePath = "/tmp/vscodereactnative.sock";
+
+export let getPipePath = (): string => {
+    return (process.platform === "win32" ? WIN_ServerPipePath : UNIX_ServerPipePath);
+};
+
+/**
+ * Defines the messages sent to the extension.
+ * Add new messages to this enum.
+ */
+export enum ExtensionMessage {
+    START_PACKAGER,
+    STOP_PACKAGER,
+    PREWARM_BUNDLE_CACHE
+}
+
+export interface MessageWithArguments {
+    message: ExtensionMessage;
+    args: any[];
+}
+
+/**
+ * Sends messages to the extension.
+ */
+export class ExtensionMessageSender {
+
+    public sendMessage(message: ExtensionMessage, args?: any[]): Q.Promise<any> {
```

Comment:
I think that changing this method to expect an already built MessageWithArguments  instead of 2 parameters will make the code simpler.